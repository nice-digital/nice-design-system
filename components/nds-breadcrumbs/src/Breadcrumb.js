import React from "react";
import PropTypes from "prop-types";

export const Breadcrumb = props => {
	const { elementType, to, children, ...attributes } = props;

	const innerTagProps = {
		...attributes
	};

	let InnerHtmlTag = "span";

	if (to) {
		InnerHtmlTag = elementType || "a";
		innerTagProps[InnerHtmlTag === "a" ? "href" : "to"] = to;
	}

	return (
		<li className="breadcrumbs__crumb">
			<InnerHtmlTag {...innerTagProps}>{children}</InnerHtmlTag>
		</li>
	);
};

Breadcrumb.propTypes = {
	elementType: PropTypes.elementType, // Allow tag to be customised for custom routing integration
	children: PropTypes.string.isRequired,
	to: PropTypes.string
};

Breadcrumb.defaultProps = {
	elementType: "a"
};
