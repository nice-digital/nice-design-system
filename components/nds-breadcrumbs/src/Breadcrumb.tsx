import { ReactNode, ElementType } from "react";
import classnames from "classnames";

export interface BreadcrumbProps {
	[prop: string]: unknown;
	children: ReactNode;
	to?: string;
	elementType?: ElementType;
	method?: string;
	className?: string;
}

export const Breadcrumb = (props: BreadcrumbProps) => {
	const { elementType, method, to, children, className, ...attributes } = props;

	let ElementType = elementType || "span";

	const innerTagProps = {
		...attributes
	};

	if (to) {
		ElementType = elementType || "a";
		innerTagProps[method || (ElementType === "a" && "href") || "to"] = to;
	}

	return (
		<li className={classnames(["breadcrumbs__crumb", className])}>
			<ElementType {...innerTagProps}>{children}</ElementType>
		</li>
	);
};
