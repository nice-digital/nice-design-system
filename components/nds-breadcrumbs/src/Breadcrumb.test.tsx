import { render } from "@testing-library/react";
import { Breadcrumb } from "./Breadcrumb";

describe("Breadcrumb", () => {
	it("should match snapshot for no location", () => {
		const wrapper = render(<Breadcrumb>Home</Breadcrumb>);
		expect(wrapper).toMatchSnapshot();
	});

	it("should match snapshot for anchor", () => {
		const wrapper = render(<Breadcrumb to="/">Home</Breadcrumb>);
		expect(wrapper).toMatchSnapshot();
	});

	it("should pass on props to rendered anchor", () => {
		const wrapper = render(
			<Breadcrumb to="/" aria-current={true}>
				Home
			</Breadcrumb>
		);

		const link = wrapper.getByRole("link");
		expect(link.getAttribute("href")).toBe("/");
		expect(link.getAttribute("aria-current")).toBe("true");
		expect(link.textContent).toBe("Home");
	});

	it("should merge any additional classes onto the container", () => {
		const wrapper = render(
			<Breadcrumb to="/" aria-current={true} className="mt--0">
				Home
			</Breadcrumb>
		);

		expect(wrapper.getByRole("listitem")).toHaveClass("mt--0");
	});
});
