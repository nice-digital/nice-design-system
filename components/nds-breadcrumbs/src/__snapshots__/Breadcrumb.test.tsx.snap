// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb should match snapshot for anchor 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="/"
        >
          Home
        </a>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="breadcrumbs__crumb"
    >
      <a
        href="/"
      >
        Home
      </a>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Breadcrumb should match snapshot for no location 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="breadcrumbs__crumb"
      >
        <span>
          Home
        </span>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="breadcrumbs__crumb"
    >
      <span>
        Home
      </span>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
